### ISSUE TITLE
Release resources allocated with ResourceLoader::loadResources

### ISSUE THREAD

**Issue by elisemorysc** [google/filament#3888]
**Describe the bug**
Not really sure if this is a bug, or a problem in our code.
We're trying to render a large number of gltf models, one at a time.

Code looks like this:
```
auto asset = assetLoader->createAssetFromBinary(
        reinterpret_cast<const uint8_t*>(gltfData.c_str()), gltfData.length());
    resourceLoader->loadResources(asset);
    scene->addEntities(asset->getEntities(), asset->getEntityCount());
    
    std::vector<uint8_t> outputImage(width * height * 4);
    backend::PixelBufferDescriptor pd(
        outputImage.data(), outputImage.size(), backend::PixelDataFormat::RGBA,
        backend::PixelDataType::UBYTE, nullptr, nullptr);
    
    renderer->beginFrame(swapchain);
    renderer->render(view);
    renderer->readPixels(0, 0, width, height, std::move(pd));
    renderer->endFrame();
    engine->flushAndWait();
    ...
    <save the buffer into a file>
    ...
    scene->removeEntities(asset->getEntities(), asset->getEntityCount());
    assetLoader->destroyAsset(asset);
```

This runs in a loop, receiving new models and render. It works as expected, however it looks like the resources loaded by `resourceLoader->loadResources(asset)` are not released and memory is "leaking". So after rendering several thousand models (all different), we run out of memory.

Is there a missing step here required to release these resources?

**Comment by romainguy** [comment]
`destroyAsset()` should be enough. Checking the code it should be destroying everything. Are you sure the resources loaded by the `ResourceLoader` is what's leaking? The only thing not destroyed are the materials (the material *instances* are destroyed), but there's only a finite number of those that can possibly be created (you could call `AssetLoader::getMaterialsCount()` to check how many have been created, maybe a bug there?).

**Comment by elisemorysc** [comment]
Hi. Thanks for the fast reply.
I think this is related to ResourceLoader since removing this will eliminate the leak. Material count doesn't change (stay at 3)

Example of a minimal version showing this: 
```
auto asset = assetLoader->createAssetFromBinary(
        reinterpret_cast<const uint8_t*>(gltfData.c_str()), gltfData.length());
    resourceLoader->loadResources(asset); // **remove this line to solve the leak**
    renderer->beginFrame(swapchain);
    renderer->render(view);
    renderer->endFrame();
    engine->flushAndWait();
    assetLoader->destroyAsset(asset); 
```
(obviously will render nothing, but illustrate the problem)
You can use the same model, you don't need to supply a different one. 
I think the problem is somehow related to `FVertexBuffer::setBufferObjectAt`. If I just return the function without assigning the buffer, it's not leaking.
Is there a way to force release these resources?

**Comment by prideout** [comment]
I'll see if I can repro, maybe we introduced a leak when we switched over to the BufferObject API.  At first glance the BufferObjects _are_ being destroyed, but maybe something more subtle is going on.

**Comment by elisemorysc** [comment]
Hi,
Memory leak was fixed in #3894. However I think it's still leaking some resources.
`OpenGLDriver::createVertexBufferS()` allocated the `GLVertexBuffer` object, which in turn allocate memory in `OpenGLDriver::HandleAllocator::alloc`
However when releasing the `GLVertexBuffer` object in `OpenGLDriver::destroyVertexBuffer` the code to release memory in `OpenGLDriver::HandleAllocator::free` was removed.

After a while the pool used for `GLVertexBuffer` (`mPool2`) will run out of nodes and the program will crash.
